src/infinite-list/index.tsx

```diff
@@ -157,13 +162,17 @@ const InfiniteList = (props: InfiniteListProps, ref: any) => {
       isHorizontal={isHorizontal}
       rowRenderer={renderItem}
       dataProvider={dataProvider}
-      layoutProvider={layoutProvider.current}
+      layoutProvider={props.layoutProvider ?? layoutProvider.current}
       extendedState={extendedState}
       initialRenderIndex={initialPageIndex}
       renderAheadOffset={5 * pageWidth}
       onScroll={onScroll}
       style={style}
       scrollViewProps={scrollViewPropsMemo}
+      onMomentumScrollEnd={scrollViewProps?.onMomentumScrollEnd}
```

Comment:
scrollViewPropsMemo should include 'onMomentumScrollEnd', why separate them?

Reply:
removed